design-documents/graph-ql/coverage/customer/gift-registry.graphqls

```diff
@@ -128,40 +130,63 @@ input GiftRegistryDynamicAttributeInput {
     value: String!
 }
 
-type CreateGiftRegistryOutput {
-    gift_registry: GiftRegistry
+interface GiftRegistryUserErrorsInterface {
+    status: Boolean!
+    user_errors: [GiftRegistryItemsUserError]!
+}
+
+interface GiftRegistryItemUserErrorInterface {
+    status: Boolean!
+    user_errors: GiftRegistryItemsUserError
```

Comment:
I don't like having a `GiftRegistryItemsUserError` type that doesn't implement `GiftRegistryItemUserErrorInterface`, that's a bad naming convention.

Reply:
I would rename `GiftRegistryItemsUserError` to `GiftRegistryItemError`; not all error types are due to user input (such as DB connection failure).

Reply:
Similarly I would rename the `user_errors` field to just `errors`.  Also it should be singular in this interface as we only return one.

Reply:
> I don't like having a `GiftRegistryItemsUserError` type that doesn't implement the `GiftRegistryItemUserErrorInterface` interface that we also have, that's a bad naming convention.

Yes that's a good point, it's a lower hierarchy. Would only apply to the same level or hierarchy if GiftRegistryItemsUserError type that doesn't implement the GiftRegistryItemUserErrorInterface

The term User doesn't come because the user caused an error. It comes from the fact that we have some error to output to the user. However the user here is PWA
And if we look what's inside, information is mixed, some of the info like code:OUT_OF_STOCK has to be processed to be output.

```
type GiftRegistryItemsUserError {
    message: String! - output for the user
    product_uid: ID - not output for the user
    gift_registry_item_uid: ID not output for the user
    gift_registry_uid: ID not output for the user
    code: GiftRegistryItemsUserErrorType! - output for the user but after translated by PWA
  }

```